In [56]:
import sys
import os

project_root = os.path.abspath('..')
if project_root not in sys.path:
    sys.path.append(project_root)

%load_ext autoreload
%autoreload 2


The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In [57]:
import pandas as pd
import numpy as np
import polars as pl

import matplotlib.pyplot as plt
import seaborn as sns

from src.data.make_dataset import DICT_VARIABLES
from src.features import build_features

sns.set_style()
plt.rcParams["figure.figsize"] = (12, 6)

HOSPITAL_A_ANALIZAR = 112103  # Este es el Torax

In [71]:
cie_9 = pl.read_excel(
    "../data/external/Esquema_Registro-2023.xlsx",
    sheet_name="Anexo 2",
    read_csv_options={"skip_rows": 5},
)

cie_10 = pl.read_excel("../data/external/CIE-10 - sin_puntos_y_X.xlsx")

In [72]:
cie_10

shape: (39_874, 5)
┌────────┬──────────────────────┬──────────────────────┬─────────────────────┬─────────────────────┐
│ Código ┆ Descripción          ┆ Categoría            ┆ Sección             ┆ Capítulo            │
│ ---    ┆ ---                  ┆ ---                  ┆ ---                 ┆ ---                 │
│ str    ┆ str                  ┆ str                  ┆ str                 ┆ str                 │
╞════════╪══════════════════════╪══════════════════════╪═════════════════════╪═════════════════════╡
│ A00X   ┆ Cólera               ┆ A00 CÓLERA           ┆ A00-A09             ┆ Cap.01  CIERTAS     │
│        ┆                      ┆                      ┆ ENFERMEDADES        ┆ ENFERMEDADES INF…   │
│        ┆                      ┆                      ┆ INFECCIOSA…         ┆                     │
│ A000   ┆ Cólera debido a      ┆ A00 CÓLERA           ┆ A00-A09             ┆ Cap.01  CIERTAS     │
│        ┆ Vibrio cholerae …    ┆                      ┆ ENFERMEDADES        ┆ ENFERMEDADES INF…   │
│        ┆                      ┆                      ┆ INFECCIOSA…         ┆                     │
│ A001   ┆ Cólera debido a      ┆ A00 CÓLERA           ┆ A00-A09             ┆ Cap.01  CIERTAS     │
│        ┆ Vibrio cholerae …    ┆                      ┆ ENFERMEDADES        ┆ ENFERMEDADES INF…   │
│        ┆                      ┆                      ┆ INFECCIOSA…         ┆                     │
│ A009   ┆ Cólera, no           ┆ A00 CÓLERA           ┆ A00-A09             ┆ Cap.01  CIERTAS     │
│        ┆ especificado         ┆                      ┆ ENFERMEDADES        ┆ ENFERMEDADES INF…   │
│        ┆                      ┆                      ┆ INFECCIOSA…         ┆                     │
│ …      ┆ …                    ┆ …                    ┆ …                   ┆ …                   │
│ Z994   ┆ Dependencia de       ┆ Z99 DEPENDENCIA DE   ┆ Z77-Z99  PERSONAS   ┆ Cap.21  FACTORES    │
│        ┆ corazón artificia…   ┆ MÁQUINAS Y DI…       ┆ CON RIESGOS PO…     ┆ QUE INFLUYEN EN…    │
│ Z998   ┆ Dependencia de otras ┆ Z99 DEPENDENCIA DE   ┆ Z77-Z99  PERSONAS   ┆ Cap.21  FACTORES    │
│        ┆ máquinas y …         ┆ MÁQUINAS Y DI…       ┆ CON RIESGOS PO…     ┆ QUE INFLUYEN EN…    │
│ Z999   ┆ Dependencia de       ┆ Z99 DEPENDENCIA DE   ┆ Z77-Z99  PERSONAS   ┆ Cap.21  FACTORES    │
│        ┆ máquina y disposi…   ┆ MÁQUINAS Y DI…       ┆ CON RIESGOS PO…     ┆ QUE INFLUYEN EN…    │
│ XXXX   ┆ Sin especificar      ┆ Sin especificar      ┆ Sin especificar     ┆ Sin especificar     │
└────────┴──────────────────────┴──────────────────────┴─────────────────────┴─────────────────────┘

## 1. Análisis Sociodemográfico de Pacientes Atendidos en el Hospital del Tórax

En este cuadernillo se quiere ver el perfil sociodemográfico de los pacientes hospitalizados
en el Hospital del Tórax. Se utilizará la base de datos pública de egresos hospitalarios DEIS
entre 2001 y 2020. Las preguntas que se quieren responder son:

- ¿Cómo es la distribución geográfica de los pacientes atendidos en el INT?
- ¿Cuál es el rango etario más atendido en el INT?
- ¿Cuál es el género más atendido en el INT?
- ¿Cuál es la previsión más atendida en el INT?
- ¿Cuál es la letra de FONASA más atendida en el INT?

Para responder estas preguntas se realizará una agrupación de los datos con todas las variables
sociodemográficas de interés para el análisis.

In [59]:
# Define todas las variables que se quieren saber

VARIABLES_SOCIODEMOGRAFICAS_ESTATICAS = ["ANO_EGRESO", "REGION_RESIDENCIA", "COMUNA_RESIDENCIA"]
VARIABLES_SOCIODEMOGRAFICAS_DINAMICAS = ["SEXO", "EDAD_CATEGORIA", "PREVISION", "BENEFICIARIO"]
VARIABLE_A_CONTAR = "DIAG1"

agrupacion_total = (
    VARIABLES_SOCIODEMOGRAFICAS_ESTATICAS
    + VARIABLES_SOCIODEMOGRAFICAS_DINAMICAS
    + [VARIABLE_A_CONTAR]
)

Luego, se genera la consulta de la base de datos. Para reducir el coste computacional, se filtra
la base de datos únicamente para los datos del INT. Posteriormente, se agrupa la base de datos
con las variables sociodemográficas de interés y se cuenta la ocurrencia de cada grupo.

In [73]:
desglose_sociodemografico = (
    pl.scan_csv("../data/processed/egresos_procesados.csv")
    .filter(pl.col("ESTABLECIMIENTO_SALUD") == HOSPITAL_A_ANALIZAR)
    .group_by(agrupacion_total)
    .count()
    .sort(["ANO_EGRESO"])
    .collect(streaming=True)
)

In [74]:
desglose_sociodemografico_cie = (
    desglose_sociodemografico.join(
        cie_9, how="left", left_on="COMUNA_RESIDENCIA", right_on="Código Comuna"
    )
    .with_columns(pl.col(["PREVISION", "Nombre Comuna"]).str.to_titlecase())
    .drop("Código Región")
)

desglose_sociodemografico_cie = desglose_sociodemografico_cie.join(
    cie_10, how="left", left_on="DIAG1", right_on="Código"
)

In [76]:
display(desglose_sociodemografico_cie.head())

shape: (5, 15)
┌────────────┬────────────┬───────────┬────────┬───┬───────────┬───────────┬───────────┬───────────┐
│ ANO_EGRESO ┆ REGION_RES ┆ COMUNA_RE ┆ SEXO   ┆ … ┆ Descripci ┆ Categoría ┆ Sección   ┆ Capítulo  │
│ ---        ┆ IDENCIA    ┆ SIDENCIA  ┆ ---    ┆   ┆ ón        ┆ ---       ┆ ---       ┆ ---       │
│ i64        ┆ ---        ┆ ---       ┆ str    ┆   ┆ ---       ┆ str       ┆ str       ┆ str       │
│            ┆ i64        ┆ i64       ┆        ┆   ┆ str       ┆           ┆           ┆           │
╞════════════╪════════════╪═══════════╪════════╪═══╪═══════════╪═══════════╪═══════════╪═══════════╡
│ 2001       ┆ 13         ┆ 13201     ┆ Mujer  ┆ … ┆ Enfermeda ┆ J44 OTRAS ┆ J40-J47   ┆ Cap.10    │
│            ┆            ┆           ┆        ┆   ┆ d         ┆ ENFERMEDA ┆ ENFERMEDA ┆ ENFERMEDA │
│            ┆            ┆           ┆        ┆   ┆ pulmonar  ┆ DES PULMO ┆ DES       ┆ DES DEL   │
│            ┆            ┆           ┆        ┆   ┆ obstructi ┆ NARE…     ┆ CRÓNICAS  ┆ APARATO…  │
│            ┆            ┆           ┆        ┆   ┆ va …      ┆           ┆ D…        ┆           │
│ 2001       ┆ 13         ┆ 13129     ┆ Mujer  ┆ … ┆ Asma, no  ┆ J45 ASMA  ┆ J40-J47   ┆ Cap.10    │
│            ┆            ┆           ┆        ┆   ┆ especific ┆           ┆ ENFERMEDA ┆ ENFERMEDA │
│            ┆            ┆           ┆        ┆   ┆ ada       ┆           ┆ DES       ┆ DES DEL   │
│            ┆            ┆           ┆        ┆   ┆           ┆           ┆ CRÓNICAS  ┆ APARATO…  │
│            ┆            ┆           ┆        ┆   ┆           ┆           ┆ D…        ┆           │
│ 2001       ┆ 13         ┆ 13114     ┆ Mujer  ┆ … ┆ Enfermeda ┆ J44 OTRAS ┆ J40-J47   ┆ Cap.10    │
│            ┆            ┆           ┆        ┆   ┆ d         ┆ ENFERMEDA ┆ ENFERMEDA ┆ ENFERMEDA │
│            ┆            ┆           ┆        ┆   ┆ pulmonar  ┆ DES PULMO ┆ DES       ┆ DES DEL   │
│            ┆            ┆           ┆        ┆   ┆ obstructi ┆ NARE…     ┆ CRÓNICAS  ┆ APARATO…  │
│            ┆            ┆           ┆        ┆   ┆ va …      ┆           ┆ D…        ┆           │
│ 2001       ┆ 13         ┆ 13127     ┆ Mujer  ┆ … ┆ Examen de ┆ Z13       ┆ Z00-Z13   ┆ Cap.21    │
│            ┆            ┆           ┆        ┆   ┆ pesquisa  ┆ EXAMEN DE ┆ PERSONAS  ┆ FACTORES  │
│            ┆            ┆           ┆        ┆   ┆ especial  ┆ PESQUISA  ┆ EN        ┆ QUE       │
│            ┆            ┆           ┆        ┆   ┆ para…     ┆ ESPECIAL  ┆ CONTACTO  ┆ INFLUYEN  │
│            ┆            ┆           ┆        ┆   ┆           ┆ …         ┆ CO…       ┆ EN…       │
│ 2001       ┆ 13         ┆ 13123     ┆ Hombre ┆ … ┆ Neumonía, ┆ J18       ┆ J09-J18   ┆ Cap.10    │
│            ┆            ┆           ┆        ┆   ┆ no especi ┆ NEUMONÍA, ┆ GRIPE Y   ┆ ENFERMEDA │
│            ┆            ┆           ┆        ┆   ┆ ficada    ┆ ORGANISMO ┆ NEUMONÍA  ┆ DES DEL   │
│            ┆            ┆           ┆        ┆   ┆           ┆ NO ESPEC… ┆           ┆ APARATO…  │
└────────────┴────────────┴───────────┴────────┴───┴───────────┴───────────┴───────────┴───────────┘

En los resultados se indica el perfil sociodemográfico para cada combinación de variable presente
en la base de datos. A modo de ejemplo, hubo 1 ocurrencia en el 2001 de una paciente de la Región
número 13 y Comuna 13605 que tenía entre 70 a 80 años que tenía PREVISION FONASA Letra B diagnosticada
con el diagnóstico K810.

Uno de los diagnósticos de interés para el Instituto es el de código "C341".

In [78]:
display(desglose_sociodemografico_cie.filter(pl.col("DIAG1") == "C341"))
print(desglose_sociodemografico_cie.filter(pl.col("DIAG1") == "C341").select(pl.col("count").sum()))

shape: (1_247, 15)
┌────────────┬────────────┬───────────┬────────┬───┬───────────┬───────────┬───────────┬───────────┐
│ ANO_EGRESO ┆ REGION_RES ┆ COMUNA_RE ┆ SEXO   ┆ … ┆ Descripci ┆ Categoría ┆ Sección   ┆ Capítulo  │
│ ---        ┆ IDENCIA    ┆ SIDENCIA  ┆ ---    ┆   ┆ ón        ┆ ---       ┆ ---       ┆ ---       │
│ i64        ┆ ---        ┆ ---       ┆ str    ┆   ┆ ---       ┆ str       ┆ str       ┆ str       │
│            ┆ i64        ┆ i64       ┆        ┆   ┆ str       ┆           ┆           ┆           │
╞════════════╪════════════╪═══════════╪════════╪═══╪═══════════╪═══════════╪═══════════╪═══════════╡
│ 2001       ┆ 3          ┆ 3102      ┆ Hombre ┆ … ┆ Tumor     ┆ C34 TUMOR ┆ C30-C39   ┆ Cap.02    │
│            ┆            ┆           ┆        ┆   ┆ maligno   ┆ MALIGNO   ┆ NEOPLASIA ┆ NEOPLASIA │
│            ┆            ┆           ┆        ┆   ┆ del       ┆ DE LOS    ┆ S         ┆ S         │
│            ┆            ┆           ┆        ┆   ┆ lóbulo    ┆ BRONQUI…  ┆ MALIGNAS  ┆ (C00-D49) │
│            ┆            ┆           ┆        ┆   ┆ superio…  ┆           ┆ DE …      ┆           │
│ 2001       ┆ 13         ┆ 13101     ┆ Hombre ┆ … ┆ Tumor     ┆ C34 TUMOR ┆ C30-C39   ┆ Cap.02    │
│            ┆            ┆           ┆        ┆   ┆ maligno   ┆ MALIGNO   ┆ NEOPLASIA ┆ NEOPLASIA │
│            ┆            ┆           ┆        ┆   ┆ del       ┆ DE LOS    ┆ S         ┆ S         │
│            ┆            ┆           ┆        ┆   ┆ lóbulo    ┆ BRONQUI…  ┆ MALIGNAS  ┆ (C00-D49) │
│            ┆            ┆           ┆        ┆   ┆ superio…  ┆           ┆ DE …      ┆           │
│ 2001       ┆ 3          ┆ 3101      ┆ Hombre ┆ … ┆ Tumor     ┆ C34 TUMOR ┆ C30-C39   ┆ Cap.02    │
│            ┆            ┆           ┆        ┆   ┆ maligno   ┆ MALIGNO   ┆ NEOPLASIA ┆ NEOPLASIA │
│            ┆            ┆           ┆        ┆   ┆ del       ┆ DE LOS    ┆ S         ┆ S         │
│            ┆            ┆           ┆        ┆   ┆ lóbulo    ┆ BRONQUI…  ┆ MALIGNAS  ┆ (C00-D49) │
│            ┆            ┆           ┆        ┆   ┆ superio…  ┆           ┆ DE …      ┆           │
│ 2001       ┆ 13         ┆ 13128     ┆ Mujer  ┆ … ┆ Tumor     ┆ C34 TUMOR ┆ C30-C39   ┆ Cap.02    │
│            ┆            ┆           ┆        ┆   ┆ maligno   ┆ MALIGNO   ┆ NEOPLASIA ┆ NEOPLASIA │
│            ┆            ┆           ┆        ┆   ┆ del       ┆ DE LOS    ┆ S         ┆ S         │
│            ┆            ┆           ┆        ┆   ┆ lóbulo    ┆ BRONQUI…  ┆ MALIGNAS  ┆ (C00-D49) │
│            ┆            ┆           ┆        ┆   ┆ superio…  ┆           ┆ DE …      ┆           │
│ …          ┆ …          ┆ …         ┆ …      ┆ … ┆ …         ┆ …         ┆ …         ┆ …         │
│ 2020       ┆ 13         ┆ 13128     ┆ Hombre ┆ … ┆ Tumor     ┆ C34 TUMOR ┆ C30-C39   ┆ Cap.02    │
│            ┆            ┆           ┆        ┆   ┆ maligno   ┆ MALIGNO   ┆ NEOPLASIA ┆ NEOPLASIA │
│            ┆            ┆           ┆        ┆   ┆ del       ┆ DE LOS    ┆ S         ┆ S         │
│            ┆            ┆           ┆        ┆   ┆ lóbulo    ┆ BRONQUI…  ┆ MALIGNAS  ┆ (C00-D49) │
│            ┆            ┆           ┆        ┆   ┆ superio…  ┆           ┆ DE …      ┆           │
│ 2020       ┆ 13         ┆ 13132     ┆ Mujer  ┆ … ┆ Tumor     ┆ C34 TUMOR ┆ C30-C39   ┆ Cap.02    │
│            ┆            ┆           ┆        ┆   ┆ maligno   ┆ MALIGNO   ┆ NEOPLASIA ┆ NEOPLASIA │
│            ┆            ┆           ┆        ┆   ┆ del       ┆ DE LOS    ┆ S         ┆ S         │
│            ┆            ┆           ┆        ┆   ┆ lóbulo    ┆ BRONQUI…  ┆ MALIGNAS  ┆ (C00-D49) │
│            ┆            ┆           ┆        ┆   ┆ superio…  ┆           ┆ DE …      ┆           │
│ 2020       ┆ 13         ┆ 13119     ┆ Hombre ┆ … ┆ Tumor     ┆ C34 TUMOR ┆ C30-C39   ┆ Cap.02    │
│            ┆            ┆           ┆        ┆   ┆ maligno   ┆ MALIGNO   ┆ NEOPLASIA ┆ NEOPLASIA │
│            ┆            ┆           ┆   

shape: (1, 1)
┌───────┐
│ count │
│ ---   │
│ u32   │
╞═══════╡
│ 1709  │
└───────┘


Por lo tanto, se muestra cada desglose sociodemográfico de los pacientes atendidos. Además,
se indica que la suma de cada conteo de grupo suma 1709 entre 2001 y 2020. Esto está en concordancia
con los egresos totales (1709) para el diagnóstico mencionado en el periodo mencionado.